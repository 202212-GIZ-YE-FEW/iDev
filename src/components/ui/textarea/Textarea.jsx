export default function Textarea(props) {
    const {
        label,
        placeholder,
        size,
        labelColor = "gray",
        height = "24",
        radius = "md",
        border = "gray",
        shadow = "sm",
        ...rest
    } = props;

    const inputClasses = `
    block p-2.5 w-full px-4 py-2 text-base text-gray bg-gray
    h-${height} ${size} rounded-${radius} placeholder-light-gray text-gray bg-white resize-none
    border-${border} shadow-${shadow} focus:outline-none border-[1px]
    `;

    return (
        <>
            {label && (
                <label
                    className={`block mb-2 font-medium text-sm md:text-base lg:text-lg text-${labelColor} capitalize`}
                >
                    {label}
                </label>
            )}
            <textarea
                className={inputClasses}
                placeholder={placeholder}
                {...rest}
            />
        </>
    );
}
