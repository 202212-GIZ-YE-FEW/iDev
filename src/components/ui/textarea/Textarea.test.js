import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import Textarea from "./Textarea";

describe("Textarea component", () => {
    test("renders label and placeholder correctly", () => {
        const labelText = "Enter your message:";
        const placeholderText = "Type your message here...";
        render(<Textarea label={labelText} placeholder={placeholderText} />);

        const labelElement = screen.getByText(labelText);
        const inputElement = screen.getByPlaceholderText(placeholderText);

        expect(labelElement).toBeInTheDocument();
        expect(inputElement).toBeInTheDocument();
    });

    test("applies custom styles correctly", () => {
        const size = "base";
        const height = "12";
        const radius = "lg";
        const border = "blue";
        const shadow = "lg";
        render(
            <Textarea
                size={size}
                height={height}
                radius={radius}
                border={border}
                shadow={shadow}
            />
        );

        const inputElement = screen.getByRole("textbox");

        expect(inputElement).toHaveClass(`text-${size}`);
        expect(inputElement).toHaveClass(`h-${height}`);
        expect(inputElement).toHaveClass(`rounded-${radius}`);
        expect(inputElement).toHaveClass(`border-${border}`);
        expect(inputElement).toHaveClass(`shadow-${shadow}`);
    });
});

it("renders correctly", () => {
    const tree = renderer.create(<Textarea />).toJSON();
    expect(tree).toMatchSnapshot();
});
