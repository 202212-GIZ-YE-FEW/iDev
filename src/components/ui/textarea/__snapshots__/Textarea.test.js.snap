// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<textarea
  className="
    block p-2.5 w-full px-4 py-2 text-base text-gray bg-gray
    h-24 undefined rounded-md placeholder-light-gray text-gray bg-white resize-none
    border-light-gray/30 shadow-sm border-[1px]
    "
/>
`;
